pupyl/indexer/facets.py

```diff
@@ -162,18 +162,39 @@ def append(self, tensor):
         ----------
         tensor: numpy.ndarray or list
             A vector to insert into index.
+
+        check_unique (optional, default: False): bool
+            Defines if append method should verify the existence
+            of a really similar tensor on the current index. In other words,
+            it checks for the unicity of the value. Be advised that this check
+            creates an overhead on the append process.
         """
         if sum(tensor) == 0.:
             raise NullTensorError
 
         if self._is_new_index:
+            if check_unique and len(self) > 1:
+
+                self.tree.build(self.size << intmul >> self.trees)
+
+                result = self.item(
+                    self.index(tensor),
+                    top=1,
+                    distances=True
+                )
+
+                if result[1][0] == 0.:
```

Comment:
Is this comparison prone to false positives? I mean, two images very similar (but not identical) to get a distance of 0? The consequence in this case would be to raise a UniqueItemError when actually that's not the case. If it is prone to false positives, how should we deal with that?

Reply:
You're right @92amartins, it makes all the sense. Empirically testing, a difference of 0.05 should be enough to mitigate false positives (regarding similarity). Maybe on future releases, this parameter should be adaptive.

For instance, the images below have a difference of 0.053273506462574005
![anitta](https://user-images.githubusercontent.com/827563/102894778-62bcc200-4442-11eb-9a40-45574480bb86.jpg)
, being the only tiny change an artificial artefact on the neck accessory.

I updated the code so you can see and test the changes.